src/sources/logstash.rs

```diff
@@ -120,22 +120,29 @@ impl TcpSource for LogstashSource {
         }
     }
 
-    fn build_acker(&self, frame: &Self::Item) -> Self::Acker {
-        LogstashAcker::new(frame)
+    fn build_acker(&self, frames: &[Self::Item]) -> Self::Acker {
+        LogstashAcker::new(frames)
     }
 }
 
 struct LogstashAcker {
-    protocol: LogstashProtocolVersion,
-    sequence_number: u32,
+    // TODO: this is very likely overkill, since there are only two protocol versions and it seems
```

Comment:
Something you intend to resolve in this PR? If not, I'd be satisfied to see an issue created, referenced in this comment block.

Reply:
Good question. I don't think it's necessary to address, but I was hoping that someone with more knowledge of the logstash protocol would be able to offer an opinion (/cc @jszwedko, since you're on much of the blame). My goal with this version was to be as defensive as possible, but that doesn't lead to the most straightforward implementation and I'd prefer not to leave something that's misleading (i.e. implies that we'll ever actually have mixed protocol versions, if we won't).

Reply:
I agree it'd be unlikely, but is possible. I'd be OK disallowing mixed protocols on a single connection if it would simplify things here, but we should have an explicit assertion that it never changes.

I think you could probably simplify this to just be a tuple of version number and sequence number rather than storing the sequence numbers per protocol version. I think we can assume, even if multiple protocol versions are being used, that a single TCP stream represents one stream of events (rather than a stream per protocol version).